independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/jaxrs/ContainerResponseContextImpl.java

```diff
@@ -134,14 +134,18 @@ public Builder getLinkBuilder(String relation) {
         return response().getLinkBuilder(relation);
     }
 
+    protected Response transientResponse() {
+        return context.getResponse().transientGet();
+    }
+
     @Override
     public boolean hasEntity() {
-        return response().hasEntity();
+        return transientResponse().hasEntity();
     }
 
     @Override
     public Object getEntity() {
-        return response().getEntity();
+        return transientResponse().getEntity();
```

Comment:
The tricky part is that it not only affects the `hasEntity()` or `getEntity()` methods, it affects every method that calls `response()` LazyResponse, eg. I face this issue with `getHeaders()`.

While debugging this, I found this comment:
https://github.com/quarkusio/quarkus/blob/3a27b803b75c5ad89b173610d847c6ee5ff7cc5c/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResponseHandler.java#L154-L155

Seems that here is where the status code is set to 200.

Reply:
> eg. I face this issue with getHeaders().

Yes I know, hence the draft status of the PR